src/pages/permissions/actions/BulkDeleteGroupsBtn.tsx

```diff
@@ -0,0 +1,96 @@
+import { FC, useState } from "react";
+import { Button, useNotify } from "@canonical/react-components";
+import { LxdGroup } from "types/permissions";
+import { useToastNotification } from "context/toastNotificationProvider";
+import { useQueryClient } from "@tanstack/react-query";
+import { deleteGroups } from "api/auth-groups";
+import { queryKeys } from "util/queryKeys";
+import DeleteGroupModal from "./DeleteGroupModal";
+import { pluralize } from "util/instanceBulkActions";
+
+interface Props {
+  groups: LxdGroup[];
+  onDelete: () => void;
+  className?: string;
+}
+
+const BulkDeleteGroupsBtn: FC<Props> = ({ groups, className, onDelete }) => {
+  const notify = useNotify();
+  const toastNotify = useToastNotification();
+  const queryClient = useQueryClient();
+  const [confirming, setConfirming] = useState(false);
+  const [confirmInput, setConfirmInput] = useState("");
+  const [disableConfirm, setDisableConfirm] = useState(true);
+  const [submitting, setSubmitting] = useState(false);
+  const confirmText = "confirm-delete-group";
+
+  const handleConfirmDelete = () => {
+    setConfirming(true);
+  };
+
+  const handleCloseConfirm = () => {
+    setConfirming(false);
+  };
+
+  const handleDeleteGroups = () => {
+    setSubmitting(true);
+    deleteGroups(groups.map((group) => group.name))
+      .then(() => {
+        void queryClient.invalidateQueries({
+          predicate: (query) => {
+            return [queryKeys.identities, queryKeys.authGroups].includes(
+              query.queryKey[0] as string,
+            );
+          },
+        });
+        toastNotify.success(`${groups.length} groups deleted.`);
+      })
+      .catch((e) => {
+        // setLoading(false);
+        notify.failure("Groups deletion failed", e);
+      })
+      .finally(() => {
+        onDelete();
```

Comment:
this is probably the wrong behaviour in the error case (catch above).

Reply:
yes `onDelete` should be executed only in the success case.